### PR TITLE
Support IDPs other that Dex

### PR DESCRIPTION
**Pull request by krasish**
**Description**
As the removal of the dex component was planned, the first step in that direction is to support other Identity Providers for the OIDC flows used in the compass UI. This PR prepares compass for working with external IDPs and makes needed chart adaptations for the relevant changes in the [Compass Console](https://github.com/kyma-incubator/compass-console/pull/56).

Changes proposed in this pull request:
- Adapt the secrets and configurations for the compass Console
- Support single values in the "groups" JWT claim
- Add all configurations needed for the tokens from an external IDP to work
- Support passing OIDC provider and client-id in run.sh script for local dev

**Pull Request status**

- [X] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/gateway/templates/virtualservice.yaml [hunk 1]
```diff
@@ -90,6 +90,7 @@ spec:
       corsPolicy:
         allowOrigins:
           - regex: "(http|https)://(compass-dev|compass|compass-mf).{{ .Values.global.ingress.domainName }}(:[0-9]{4})?"
+          - regex: "(http|https)://localhost(:[0-9]{4})?"
```
Comment (alextargov) [inline chart/compass/charts/gateway/templates/virtualservice.yaml:93]: I think ports go all the way till 65535. So maybe `[0-9]{5}` would be preferable?

File: installation/cmd/run.sh [hunk 2]
```diff
@@ -113,6 +125,21 @@ function mount_minikube_ca_to_oathkeeper() {
  -p '{"spec":{"template":{"spec":{"containers":[{"name": "'$OATHKEEPER_CONTAINER_NAME'","volumeMounts": [{ "mountPath": "'/etc/ssl/certs/mk-ca.crt'","name": "minikube-ca-volume","subPath": "mk-ca.crt"}]}]}}}}'
 }
 
+function set_oidc_config() {
```
Comment (alextargov) [inline installation/cmd/run.sh:128]: As far as I understand, the `.global.cockpit.auth.*` values will be used in the in-cluster compass-console. Wouldn't it be a good idea to mention that we should set the respective flags when executing the installation script [here](https://github.com/kyma-incubator/compass/blob/main/docs/compass/04-01-installation.md#local-minikube-installation)?

File: chart/compass/charts/cockpit/templates/secret.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+type: Opaque
+stringData:
+  secret.js: |
+    if (window.clusterConfig) {
+      window.clusterConfig.idpUrl = '{{ .Values.global.cockpit.auth.idpHost }}';
+      window.clusterConfig.oidcScopes = '{{ .Values.global.cockpit.auth.scopes }}';
+      window.clusterConfig.oidcClientID = '{{ .Values.global.cockpit.auth.clientID }}';
+    } else {
+      window.clusterConfig = {
+      idpUrl: '{{ .Values.global.cockpit.auth.idpHost }}',
```
Comment (alextargov) [inline chart/compass/charts/cockpit/templates/secret.yaml:18]: Can we indent thees three props with two spaces for example

File: installation/cmd/run.sh (hunk 2 above)
Comment (krasish) [reply installation/cmd/run.sh:128]: Absolutely! I'd forgotten that.

**Review by alextargov — DISMISSED** [review]

File: installation/cmd/run.sh [hunk 4]
```diff
@@ -113,10 +153,30 @@ function mount_minikube_ca_to_oathkeeper() {
...
-    trap revert_migrator_file EXIT
+if [[ -z ${OIDC_HOST} || -z ${OIDC_CLIENT_ID} ]]; then
+  if [[ -f ${PATH_TO_COMPASS_OIDC_CONFIG_FILE} ]]; then
+    echo -e "${YELLOW}OIDC configuration not provided. Configuration from default config file will be used.${NC}"
+    OIDC_HOST=$(yq ".idpHost" "$PATH_TO_COMPASS_OIDC_CONFIG_FILE")
+    OIDC_CLIENT_ID=$(yq ".clientID" "$PATH_TO_COMPASS_OIDC_CONFIG_FILE")
+    OIDC_GROUPS=$(yq ".adminGroupNames" "$PATH_TO_COMPASS_OIDC_CONFIG_FILE")
+    set_oidc_config "$OIDC_HOST" "$OIDC_CLIENT_ID" "$OIDC_GROUPS"
+  else
+    echo -e "${RED}OIDC configuration not provided and config file was found. JWT flows will not work!${NC}"
```
Comment (DimitarPetrov) [inline installation/cmd/run.sh:164]: ... and config file was **not** found

File: chart/compass/charts/cockpit/templates/jwks-job.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+              value: "{{ .Values.global.cockpit.auth.idpHost }}/oauth2/certs"
+          command:
+            - bash
+            - -c
+            - |
+              set -e
+
+              trap "echo '# KILLING PILOT-AGENT #'; curl -X POST http://127.0.0.1:15020/quitquitquit; sleep 3;" RETURN EXIT INT TERM
+
+              wget https://github.com/mikefarah/yq/releases/download/$YQ_VERSION/yq_linux_amd64 -O /usr/bin/yq && chmod +x /usr/bin/yq
```
Comment (DimitarPetrov) [inline chart/compass/charts/cockpit/templates/jwks-job.yaml:79]: hardcoded amd64 probably will not work locally for M1 macs if the bitnami/kubectl:1.22.4 image has an arm64 version of the image.

Comment (DimitarPetrov) [reply chart/compass/charts/cockpit/templates/jwks-job.yaml:79]: consider removing the whole script if it is not needed anymore

Comment (krasish) [reply chart/compass/charts/cockpit/templates/jwks-job.yaml:79]: bitnami/kubectl:1.22.4 image still hasn't got an arm64 image.

**Review by nkanchev — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]

**Review by alextargov — APPROVED** [review]

**Pull request merged**